metadata-integration/java/datahub-schematron/lib/src/test/java/io/datahubproject/schematron/converters/avro/AvroSchemaConverterTest.java

```diff
@@ -0,0 +1,942 @@
+        "string",
+        false,
+        false,
+        new SchemaFieldDataType().setType(SchemaFieldDataType.Type.create(new StringType())));
+    assertSchemaField(
+        schema.getFields().get(2),
+        "[version=2.0].[type=StructType].[type=ComplexStruct].structField.[type=int].fieldInt",
+        "int",
+        false,
+        false,
+        new SchemaFieldDataType().setType(SchemaFieldDataType.Type.create(new NumberType())));
+    assertSchemaField(
+        schema.getFields().get(3),
+        "[version=2.0].[type=StructType].[type=ComplexStruct].structField.[type=boolean].fieldBoolean",
+        "boolean",
+        false,
+        false,
+        new SchemaFieldDataType().setType(SchemaFieldDataType.Type.create(new BooleanType())));
+    assertSchemaField(
+        schema.getFields().get(4),
+        "[version=2.0].[type=StructType].[type=ComplexStruct].structField.[type=map].fieldMap",
+        "map<string,string>",
+        false,
+        false,
+        new SchemaFieldDataType()
+            .setType(
+                SchemaFieldDataType.Type.create(
+                    new MapType().setKeyType("string").setValueType("string"))));
+    assertSchemaField(
+        schema.getFields().get(5),
+        "[version=2.0].[type=StructType].[type=ComplexStruct].structField.[type=NestedRecord].fieldRecord",
+        "NestedRecord",
+        false,
+        false,
+        new SchemaFieldDataType().setType(SchemaFieldDataType.Type.create(new RecordType())));
+    assertSchemaField(
+        schema.getFields().get(6),
+        "[version=2.0].[type=StructType].[type=ComplexStruct].structField.[type=NestedRecord].fieldRecord.[type=string].nestedField1",
+        "string",
+        false,
+        false,
+        new SchemaFieldDataType().setType(SchemaFieldDataType.Type.create(new StringType())));
+    assertSchemaField(
+        schema.getFields().get(7),
+        "[version=2.0].[type=StructType].[type=ComplexStruct].structField.[type=NestedRecord].fieldRecord.[type=int].nestedField2",
+        "int",
+        false,
+        false,
+        new SchemaFieldDataType().setType(SchemaFieldDataType.Type.create(new NumberType())));
+    assertSchemaField(
+        schema.getFields().get(8),
+        "[version=2.0].[type=StructType].[type=ComplexStruct].structField.[type=array].fieldArray",
+        "array(string)",
+        false,
+        false,
+        new SchemaFieldDataType()
+            .setType(
+                SchemaFieldDataType.Type.create(
+                    new ArrayType().setNestedType(new StringArray("string")))));
+    assertSchemaField(
+        schema.getFields().get(9),
+        "[version=2.0].[type=StructType].[type=ComplexStruct].structField.[type=union].fieldUnion",
+        "union",
+        true,
+        false,
+        new SchemaFieldDataType()
+            .setType(
+                SchemaFieldDataType.Type.create(
+                    new UnionType().setNestedTypes(new StringArray("union")))));
```

Comment:
considering this comes from 

```
          {
            "name": "fieldUnion",
            "type": [
              "null",
              "string",
              "int"
            ]
          },
```

I would expect something like

```suggestion
    assertSchemaField(
        schema.getFields().get(9),
        "[version=2.0].[type=StructType].[type=ComplexStruct].structField.[type=union].fieldUnion",
        "union<string, int>",
        true,
        false,
        new SchemaFieldDataType()
            .setType(
                SchemaFieldDataType.Type.create(
                    new UnionType().setNestedTypes(new StringArray("string", "int")))));
```

Reply:
Shouldn't this be `union<null, string, int>` as native type as it is nullable?

Reply:
line 508 is about nullable flag
but yes, `union<null, string, int>` would also look good to me

Reply:
yes, but I feel like in Avro this is the actual native type